pkg/webhook/admission/validator.go

```diff
@@ -0,0 +1,169 @@
+package admission
+
+import (
+	"fmt"
+
+	ovirtsdk "github.com/ovirt/go-ovirt"
+)
+
+const (
+	// NicInterfaceCheckID defines an ID of a NIC interface model check
+	NicInterfaceCheckID = CheckID("nic.interface")
+	// NicOnBootID defines an ID of a NIC on_boot == fales check
+	NicOnBootID = CheckID("nic.on_boot")
+	// NicPluggedID defines an ID of a NIC plugged == false check
+	NicPluggedID = CheckID("nic.plugged")
+	// NicVNicPortMirroringID defines an ID of a vnic_profile.port_mirroring == true check
+	NicVNicPortMirroringID = CheckID("nic.vnic_profile.port_mirroring")
+	// NicVNicPassThroughID defines an ID of a vnic_profile.pass_through == 'enabled' check
+	NicVNicPassThroughID = CheckID("nic.vnic_profile.pass_through")
+	// NicVNicCustomPropertiesID defines an ID of a vnic_profile.custom_properties presence check
+	NicVNicCustomPropertiesID = CheckID("nic.vnic_profile.custom_properties")
+	// NicVNicNetworkFilterID defines an ID of a vnic_profile.networ_filter presence check
+	NicVNicNetworkFilterID = CheckID("nic.vnic_profile.network_filter")
+	// NicVNicQosID defines an ID of a vnic_profile.qos presence check
+	NicVNicQosID = CheckID("nic.vnic_profile.qos")
+)
+
+var validInterfaceDeviceModels = map[string]*struct{}{"e1000": nil, "rtl8139": nil, "virtio": nil}
+
+// CheckID identifies validation check for Virtual Machine Import
+type CheckID string
+
+// ValidationFailure describes Virtual Machine Import validation failure
+type ValidationFailure struct {
+	// Check ID
+	ID CheckID
+	// Verbose explanation of the failure
+	Message string
+}
+
+func validateNics(nics []*ovirtsdk.Nic) []ValidationFailure {
+	var failures []ValidationFailure
+	for _, nic := range nics {
+		failures = append(failures, validateNic(nic)...)
+	}
+	return failures
+}
+
+func validateNic(nic *ovirtsdk.Nic) []ValidationFailure {
+	var results []ValidationFailure
+	var nicID = ""
+	if id, ok := nic.Id(); ok {
+		nicID = id
+	}
+
+	//TODO: Networking rule #1
+
+	if failure, valid := isValidNicInterfaceModel(nic, nicID); !valid {
+		results = append(results, failure)
+	}
+	if failure, valid := isValidNicOnBoot(nic, nicID); !valid {
+		results = append(results, failure)
+	}
+	if failure, valid := isValidNicPlugged(nic, nicID); !valid {
+		results = append(results, failure)
+	}
+
+	if vNicProfile, ok := nic.VnicProfile(); ok {
```

Comment:
IIUC, VnicProfile needs to be fetched by followLink since it is just a reference to the resource and not the entire entity

Reply:
Good one! I assume at that level that I've got everything I need - the client is expected to provide all necessary data - will pay attention that at that level.